lib/production_e2e_filter.rb

```diff
@@ -0,0 +1,64 @@
+require 'logger'
+
+module ProductionE2EFilter
+  def self.log_tag
+    return 'production_e2e_filter'
+  end
+
+  BEGIN {
+    logger = Logger.new(STDOUT)
+    begin
+      E2E_FILTER_CONF = YAML::load(ERB.new(IO.read(File.expand_path('./conf/e2e_filter_conf.yml', __FILE__))).result).with_indifferent_access
+      @emails = E2E_FILTER_CONF['PREFIXES']['EMAILS']
+      @domains = E2E_FILTER_CONF['PREFIXES']['DOMAINS']
+      logger.info({
+                    log_tag: self.log_tag,
+                    message: "Successfully initialized red_cross production e2e filter values"
+                  })
+    rescue => e
+      logger.error({
+                     log_tag: self.log_tag,
+                     message: "Unable to initialize red_cross production e2e filter due to #{e.message}"
+                   })
+    end
+  }
+
+  def is_filtering_enabled?
+    E2E_FILTER_CONF['EVENT_FILTERING_ENABLED']
+  end
+
+  def is_e2e_test_flow?(properties)
+    begin
+      filter_email?(properties[:email]) || filter_domain?(properties[:website], properties[:storeDomain]) ? true : false
+    rescue => e
+      logger.error({
+                     log_tag: self.log_tag,
+                     message: "Failed to check for production e2e filter values due to #{e.message}"
+                   })
+      false
+    end
+  end
+
+  private
+
+  def filter_email?(email)
+    @emails.each do |test_prefix|
+      if email&.start_with?(test_prefix)
+        return true
+      end
+    end
+    false
+  end
+
+  def filter_domain?(*args)
+    @domains.each do |test_prefix|
+      args.each do |field|
+        if field.start_with?(test_prefix)
```

Comment:
can be one liner (still this is ruby)
return true if field.start_with?(test_prefix)